src/main/java/seedu/address/logic/commands/CopyCommand.java

```diff
@@ -0,0 +1,178 @@
+import seedu.address.logic.util.ArgumentMultimap;
+import seedu.address.logic.util.ArgumentTokenizer;
+import seedu.address.logic.util.ParserUtil;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+
+/**
+ * Copies the details of a person in the address book.
+ */
+public class CopyCommand extends Command {
+
+    public static final String COMMAND_WORD = "copy";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Copies the details of the prefix identified\n"
+        + "Parameters: "
+        + "[" + PREFIX_NAME + "] "
+        + "[" + PREFIX_PHONE + "] "
+        + "[" + PREFIX_EMAIL + "] "
+        + "[" + PREFIX_ADDRESS + "] "
+        + "[" + PREFIX_TAG + "] "
+        + "[" + PREFIX_ASSET + "]\n"
+        + "Example: " + COMMAND_WORD + " 1 "
+        + PREFIX_NAME + " "
+        + PREFIX_PHONE + " "
+        + PREFIX_TAG + " "
+        + PREFIX_ASSET + " ";
+
+    public static final String MESSAGE_NO_PARAM = "At least one field to copy must be provided.";
+
+    private final Index index;
+    private final boolean[] info;
+
+    /**
+     * @param index of the person in the filtered person list to copy
+     * @param info list of details to copy to clipboard
+     */
+    public CopyCommand(Index index, boolean[] info) {
+        requireNonNull(index);
+
+        this.index = index;
+        this.info = info;
+    }
+
+    @Override
+    public String execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToCopy = lastShownList.get(index.getZeroBased());
+        String personDetails = copyToClipboard(personToCopy, info);
+
+        return String.format(Messages.MESSAGE_COPIED, personDetails);
+    }
+
+    /**
+     * Copies the details of {@code personToCopy} to clipboard
+     */
+    private static String copyToClipboard(Person personToCopy, boolean[] info) {
+        requireNonNull(personToCopy);
+
+        StringBuilder copiedMsg = new StringBuilder();
+
+        if (info[0]) {
+            copiedMsg.append("Name: ").append(personToCopy.getName()).append("; ");
+        }
+        if (info[1]) {
+            copiedMsg.append("Phone: ").append(personToCopy.getPhone()).append("; ");
+        }
+        if (info[2]) {
+            copiedMsg.append("Email: ").append(personToCopy.getEmail()).append("; ");
+        }
+        if (info[3]) {
+            copiedMsg.append("Address: ").append(personToCopy.getAddress()).append("; ");
+        }
+        if (info[4]) {
+            copiedMsg.append("Tags: ").append(personToCopy.getTags()).append("; ");
+        }
+        if (info[5]) {
+            copiedMsg.append("Assets: ").append(personToCopy.getAssets()).append("; ");
+        }
```

Comment:
Field names should not be added to clipboard.

Reply:
Field name removed